pandas/core/nanops.py

```diff
@@ -933,7 +937,12 @@ def reduction(
             result = getattr(values, meth)(axis)
 
         result = _wrap_results(result, dtype, fill_value)
-        return _maybe_null_out(result, axis, mask, values.shape)
+        result = _maybe_null_out(result, axis, mask, values.shape)
+
+        if datetimelike and not skipna:
+            result = _mask_datetimelike_result(result, axis, mask, orig_values)
```

Comment:
_maybe_null_out should just do this (e.g. just pass int he orig_values.dtype)

Reply:
i think it may belong in _wrap_results; not all these funcs call _maybe_null_out.  ive got a branch working on cleaning these up